src/stores/Frame/FrameStore.ts

```diff
@@ -1449,6 +1639,80 @@ export class FrameStore {
         return AST.getFrameFromFitsChan(fitsChan, checkSkyDomain);
     };
 
+    private initSpectralVsDirectionFrame = (): AST.FrameSet => {
+        const fitsChan = AST.emptyFitsChan();
+        const regOtherAxes = new RegExp(`(CTYPE|CDELT|CRPIX|CRVAL|CUNIT|NAXIS|CROTA)[5-9]`);
+        const regDirXNumber = new RegExp(`(CTYPE|CDELT|CRPIX|CRVAL|CUNIT|NAXIS|CROTA)${this.dirXNumber}`);
+        const regDirYNumber = new RegExp(`(CTYPE|CDELT|CRPIX|CRVAL|CUNIT|NAXIS|CROTA)${this.dirYNumber}`);
+        const regSpectralNumber = new RegExp(`(CTYPE|CDELT|CRPIX|CRVAL|CUNIT|NAXIS|CROTA)${this.spectralNumber}`);
+        const regStokesNumber = new RegExp(`(CTYPE|CDELT|CRPIX|CRVAL|CUNIT|NAXIS|CROTA)${this.stokesNumber}`);
+
+        for (let entry of this.frameInfo.fileInfoExtended.headerEntries) {
+            let name = entry.name;
+
+            if (name.match(regOtherAxes) || name.match(regStokesNumber) || name === "HISTORY") {
+                continue;
+            }
+
+            // Remove the stokes axis (if any), and reset axis numbers for x, y, and spectral.
+            if (name.match(regDirXNumber) && this.dirXNumber !== this.dirX) {
+                name = entry.name.replace(`${this.dirXNumber}`, `${this.dirX}`);
+            } else if (name.match(regDirYNumber) && this.dirYNumber !== this.dirY) {
+                name = entry.name.replace(`${this.dirYNumber}`, `${this.dirY}`);
+            } else if (name.match(regSpectralNumber) && this.spectralNumber !== this.spectral) {
+                name = entry.name.replace(`${this.spectralNumber}`, `${this.spectral}`);
+            }
+
+            if (!entry.value.length) {
+                continue;
+            }
+
+            let value = trimFitsComment(entry.value);
+            if (name.toUpperCase() === "NAXIS" || name.toUpperCase() === "WCSAXES") {
+                value = "3";
+            }
+
+            if (entry.entryType === CARTA.EntryType.STRING) {
+                value = `'${value}'`;
+            } else {
+                value = FrameStore.ShiftASTCoords(entry, value);
+            }
+
+            while (name.length < 8) {
+                name += " ";
+            }
+
+            const entryString = `${name}=  ${value}`;
+            AST.putFits(fitsChan, entryString);
+        }
+        return AST.getFrameFromFitsChan(fitsChan, false);
+    };
+
+    public updateSpectralVsDirectionWcs = () => {
+        if (this.wcsInfo3D) {
+            this.wcsInfo = AST.makeSwappedFrameSet(this.wcsInfo3D, this.dirAxis, this.spectral, this.requiredChannel, this.dirAxisSize);
+            AST.set(this.wcsInfo, `Format(${this.dirAxis})=${this.dirAxisFormat}, Unit(${this.dirAxis})=""`);
+        }
+    };
+
+    private getDirAxisInfo() {
```

Comment:
I would prefer we name it `updateDirAxisInfo`. also prefer using an arrow function.